server/src/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -997,7 +999,9 @@ private void checkPodAttributes(final long podId, final String podName, final lo
         checkIpRange(startIp, endIp, cidrAddress, cidrSize);
 
         // Check if the IP range overlaps with the public ip
-        checkOverlapPublicIpRange(zoneId, startIp, endIp);
+        if(startIp != null) {
```

Comment:
Consider changing this to `Strings.isNullOrEmpty()`

Reply:
Replaced this with the mentioned way. Also replaced in many other places. Please have a look once.